oauth2_provider/forms.py

```diff
@@ -8,3 +8,4 @@ class AllowForm(forms.Form):
     client_id = forms.CharField(widget=forms.HiddenInput())
     state = forms.CharField(required=False, widget=forms.HiddenInput())
     response_type = forms.CharField(widget=forms.HiddenInput())
+    pkce = forms.CharField(required=False, widget=forms.HiddenInput())
```

Comment:
Why is this a `CharField`? Wouldn't a boolean be more appropriate?

Reply:
This field is where the hashed (or plainext) PKCE value is sent, if the field is present we assume that the whole auth flow is using PKCE and store the value in the grant.

Reply:
I'll re-read the spec and that article I originally read, but my understanding of PKCE is that this won't work. The value is random and will change with every authentication, so storing it here won't let more than 1 person auth at once.